### DISCUSSION TITLE
Fixing  'ALPR_adapter' object has no attribute '_num_items_found'

### DISCUSSION THREAD

**Discussion by alarmatwork** [codeproject/CodeProject.AI-Server#238] [Q&A]
Whenever I try ALPR module from the the Explorer Iöm gettings folloding error:
`
13:00:46:Response rec'd from License Plate Reader command 'alpr' (...698964)
13:00:46:License Plate Reader:  [AttributeError] : Error during main_loop: Traceback (most recent call last):
  File "C:\Program Files\CodeProject\AI\modules\ALPR\bin\windows\python39\venv\lib\site-packages\codeproject_ai_sdk\module_runner.py", line 618, in main_loop
    self.update_statistics(output)
  File "C:\Program Files\CodeProject\AI\modules\ALPR\ALPR_adapter.py", line 92, in update_statistics
    self._num_items_found += len(predictions)
AttributeError: 'ALPR_adapter' object has no attribute '_num_items_found'
`

I'm running  Module 'License Plate Reader' 3.3.4 (ID: ALPR)

**Comment by ChrisMaunder** [comment]
In the ALPR_adapter.py file, do you see this?

<img width="1163" alt="Screenshot 2024-12-19 at 10 07 05 am" src="https://github.com/user-attachments/assets/b00448e0-4863-4474-907e-85fbbd4e98e7" />

**Reply by alarmatwork** [reply]
Yes, I can see similar code. What should I change ?

**Reply by wired64** [reply]
I also have the same error - with the same ALPR version.  

I can see line 44 setting the _num_items_found variable.  I can also see that it's used in two function definitions (status and update_statistics.  

I'm not a refined Python developer (yet), but the use of the variable appears to be correct and there don't appear to be any finger fumble spelling mistakes - here are the lines of code where I see it used):

    def status(self) -> JSON:
        statusData = super().status()
        statusData["numItemsFound"] = self._num_items_found
        return statusData


    def update_statistics(self, response):
        super().update_statistics(response)
        if "success" in response and response["success"] and "predictions" in response:
            predictions = response["predictions"]
            self._num_items_found += len(predictions)

**Reply by wired64** [reply]
@alarmatwork - so I had the same problem as you - I started looking at options to resolve this and it seems that CUDA library version 11.7 is needed.  I downgraded the CUDA library to version 11.7 and voila!
